src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,200 @@
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get the car by id" + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversByCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT * FROM cars c JOIN manufacturers m ON "
+                + "c.manufacturer_id = m.id WHERE c.is_deleted = FALSE;";
+        Car car;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                car = getCar(resultSet);
+                carList.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all data from db", e);
+        }
+        for (Car carFromList : carList) {
+            carFromList.setDrivers(getDriversByCar(carFromList.getId()));
+        }
+        return carList;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars c SET model = ?, manufacturer_id = ? "
+                + "WHERE c.id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement = connection.prepareStatement(query)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update car " + car, e);
+        }
+        deleteDriversFromCar(car.getId());
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return (statement.executeUpdate() > 0);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete data by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT c.id, model, manufacturer_id, m.id, m.name, m.country "
+                + "FROM cars c "
+                + "LEFT JOIN cars_drivers c_d ON car_id = c.id "
```

Comment:
this is a common style for aliases:
```suggestion
                + "LEFT JOIN cars_drivers cd ON car_id = c.id "
```